snapcraft/internal/meta/plugs.py

```diff
@@ -54,15 +54,15 @@ def validate(self) -> None:
     def from_dict(cls, *, plug_dict: Dict[str, Any], plug_name: str) -> "Plug":
         """Create plug from dictionary."""
 
-        interface = plug_dict.get("interface", None)
-        if interface is not None:
-            plug_class = PLUG_MAPPINGS.get(interface, None)
-            if plug_class is not None:
-                return plug_class.from_dict(plug_dict=plug_dict, plug_name=plug_name)
+        iface = plug_dict.get("interface", plug_name)
```

Comment:
may I ask why the variable "interface" was renamed to iface?

Reply:
@sergiusens 
before i had a very long line. Now i can rename it back.

Please, don't merge it yet, because there is still a bug.

Reply:
Had renamed iface back to interface